bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -371,14 +379,24 @@ WebViewEnvironment createEnvironment() {
 
 @Override
 public void create(Composite parent, int style) {
-	checkDeadlock();
+	// If leads to deadlock then schedule the execution in a different thread and move on.
+	// The webview calls are queued to be executed when it is done executing the current task.
+	if (leadsToDeadlock() && !isAsyncExecution) {
+		isAsyncExecution = true;
+		browser.getDisplay().asyncExec(() -> create(parent, style));
+		return;
+	}
+
 	containingEnvironment = createEnvironment();
 
 	long[] ppv = new long[1];
 	int hr = containingEnvironment.environment().QueryInterface(COM.IID_ICoreWebView2Environment2, ppv);
 	if (hr == COM.S_OK) environment2 = new ICoreWebView2Environment2(ppv[0]);
 
 	hr = callAndWait(ppv, completion -> containingEnvironment.environment().CreateCoreWebView2Controller(browser.handle, completion));
+	if(browser.isDisposed()) {
+		return;
+	}
```

Comment:
The `browser` could be disposed right at the beginning of this method, right? A lot can happen between executing line 386 (`asyncExec`) and entering this method again (this time in an asynchronous call) so this check should happen at the very beginning of the method.
Also, if there is an early return, the variable `isAsyncExecution` will remain `true` so you probably have to set it back to `false` upon early returns _i.e._

```java
if(browser.isDisposed()) {
	isAsyncExecution = false;
	return;
}
```

Reply:
Nope, that's a wrong assumption. This if block is needed after the webView call because the call is serailized and the control flow is waiting at 396 for a return (which will happen if its called by a webview callback, if an async exec)

In some scenario, we might have already disposed the browser before the serialized exec could finish. There we won't be getting the right hr from the webview call as the return value. And which ends up in an error. So, to tackle that we need to check what the reason was (The widget was disposed). Thus, adding the check after the webview call. Adding a check in the beginning is of no use. Since this method is called in the constructor and ideally its not possible to dispose a browser before its created (only in the sync scenario, since the object is created and its internal stuff are deferred to be initialized - there we need to check after the exact point where it was deferred, which is the webview call)